### ISSUE TITLE
"Play next" button doesn't work

### ISSUE THREAD

**Issue by siku2** [nixgates/plugin.video.seren#219]
The current strategy used by the window is to seek to the end of the video ([`seekTime(getTotalTime())`](https://github.com/nixgates/plugin.video.seren/blob/217cfda80a4db03032ef15723b9d8ed3f04cd68b/resources/lib/gui/windows/playing_next.py#L68)). Unfortunately - at least in my case - Kodi doesn't play the next episode. Instead, it just keeps "playing" (the time keeps advancing) past the end of the video. Note that this issue does not occur when the player reaches the end naturally.
I guess this would warrant an issue somewhere else too but this is as good a place to start as any.

Wouldn't a call to the player's [playnext](https://codedocs.xyz/AlwinEsch/kodi/group__python___player.html#ga7c6e8d7ad526ca9732cca80fd03eceb7) method be more appropriate?
I'm just getting started with Kodi development so please excuse my ignorance in case I'm missing something.

I suspect this issue has something to do with my environment because I haven't seen anyone else mentioning it.

**Comment by nixgates** [comment]
There used a graphical glitch when using the playnext method in version 17. 

Funnily enough, I was doing some testing on this exact code yesterday and I think the graphical issue is fixed, but there were some other issues that needed to be addressed to make it work

I have a hotfix that I'm just holding off on a little to try and encompass any straggler bugs so I'll look into this and try and wrap it inside the next hot fix.

**Comment by henryjfry** [comment]
I have the same issue with play_next not workin due to the line "self.player.seekTime(self.player.getTotalTime())" and i got it to work with the following modification:


```
    def handle_action(self, action, control_id=None):
        if control_id is None:
            control_id = self.getFocusId()

        if control_id == 3001:
            self.actioned = True
#            self.player.seekTime(self.player.getTotalTime())
	    import xbmc
	    xbmc.executebuiltin('PlayerControl(BigSkipForward)')
            self.close()
        if control_id == 3002:
            self.actioned = True
            self.close()
```

**Comment by Arathonk** [comment]
Are there any updates for this problem? I face the exact same issue..
@henryjfry how do I apply your fix?

**Comment by drinfernoo** [comment]
This has been fixed on recent versions, as far as I'm aware? Have you tried recently @Arathonk?

**Comment by Arathonk** [comment]
@drinfernoo yes just clicked play next when at the end of an episode and it kept playing beyond the end of the video. Had to skip forward 10 sec in order for it to do do it automatically like it would happen if I didn't press anything.

**Comment by drinfernoo** [comment]
@Arathonk I haven't had this happen. Are you using a different skin or a different Seren theme?

On Seren 1.6.16, "Playing Next" and "Still Watching" work exactly as expected.

**Comment by henryjfry** [comment]
> @drinfernoo yes just clicked play next when at the end of an episode and it kept playing beyond the end of the video. Had to skip forward 10 sec in order for it to do do it automatically like it would happen if I didn't press anything.

Yeah thats whats would happen to me with seren play next, it would play beyond the end of the video.  To fix this:

~/.kodi/addons/plugin.video.seren/resources/lib/gui/windows/playing_next.py

`        if control_id == 3001:
            self.actioned = True
#            self.player.seekTime(self.player.getTotalTime())
	    import xbmc
	    xbmc.executebuiltin('PlayerControl(BigSkipForward)')
            self.close()
        if control_id == 3002:
            self.actioned = True
            self.close()`

So it must be a problem with "self.player.getTotalTime()"

**Comment by Arathonk** [comment]
@drinfernoo im not using a different skin or theme and I'm on Seren 1.6.16.

@henryjfry how do I apply that fix you give?
I know where to find that file and to open it but what do I do with the text you give?

**Comment by henryjfry** [comment]
> @drinfernoo im not using a different skin or theme and I'm on Seren 1.6.16.
> 
> @henryjfry how do I apply that fix you give?
> I know where to find that file and to open it but what do I do with the text you give?

If you comment out:
"self.player.seekTime(self.player.getTotalTime())"
ie # self.player.seekTime(self.player.getTotalTime())

And below it insert:
    import xbmc
    xbmc.executebuiltin('PlayerControl(BigSkipForward)')

It will do a big jump forward (ie kodi's 10 minute jump) rather than trying to seek beyond the end of the file, which mostly didnt seem to work for me and seems to be the same issue you have.

**Comment by Arathonk** [comment]
@henryjfry so it will do that big jump only when pressing the play next button? I'll try it later today.

**Comment by henryjfry** [comment]
> @henryjfry so it will do that big jump only when pressing the play next button? I'll try it later today.

Yeah, if you skip forward yourself or it plays to the end of the episode it automatically goes to the next item in the playlist.
But normally the wee seren play next episode window would trigger a seek action, with this change its the big jump instead. Which works fine for me.

Although when seren gets updated you would need to make the fix again (unless it gets incorporated into the add-on)

**Comment by Arathonk** [comment]
@henryjfry I applied the fix you gave but it still does the same thing..

**Comment by henryjfry** [comment]
> @henryjfry I applied the fix you gave but it still does the same thing..

Can you check the logs?
Its a pretty simple change and if manually jumping forward goes to the next episode then:
import xbmc
xbmc.executebuiltin('PlayerControl(BigSkipForward)')

Should work, its a builtin function so if the code runs properly it should be the same as if you hit the button on your remote.
Python needs to have the correct level of indenting, without it its not going to work properly so the first thing I would check is that the indenting matches the code snippet I added above (or "Import" is aligned with "self.actioned = True").

**Comment by Arathonk** [comment]
@henryjfry Ah I forgot to comment it out, done that now and I'm gonna test it in a moment,
Heres what it looks like now:

`        if control_id == 3001:
            self.actioned = True
#            self.player.seekTime(self.player.getTotalTime())
	    import xbmc
            xbmc.executebuiltin('PlayerControl(BigSkipForward)')
            self.close()
        if control_id == 3002:
            self.actioned = True
            self.close()`

**Comment by Arathonk** [comment]
@henryjfry 
@nixgates 

Does this fix still work in 2.0.8?

**Comment by henryjfry** [comment]
Yes, the playing next window has changed a bit but making that change "xbmc.executebuiltin('PlayerControl(BigSkipForward)')" in the equivalent section still works.

**Comment by Arathonk** [comment]
> Yes, the playing next window has changed a bit but making that change "xbmc.executebuiltin('PlayerControl(BigSkipForward)')" in the equivalent section still works.

So if i follow the same exact steps it should work?

**Comment by henryjfry** [comment]
Yep, add the BigSkipForward under "if control_id == 3001:" and it will do a big seek rather than jump to end of file.

**Comment by nixgates** [comment]
@henryjfry thanks for this! I'll implement it into the next update!

**Comment by craigsblackie** [comment]
I'm also experiencing this, I'll implement the maunal fix should this if I get the time before the next update.

**Comment by inb4after** [comment]
@nixgates a better fix is `self.seekTime(int(self.getTotalTime()))`

**Comment by nixgates** [comment]
@inb4after we currently seek using the total time of the media. 

Henry's seems to work around a faulty outside of Seren to get it working. 

@henryjfry implemented now for the next update :+1:

**Comment by inb4after** [comment]
> @inb4after we currently seek using the total time of the media. 
> 
> Henry's seems to work around a faulty outside of Seren to get it working. 
> 
> @henryjfry implemented now for the next update :+1:

Gotcha, however I was thinking by formatting the total time into the int(), any value above or below the final second will be rounded down accordingly and this will prevent the player going further.

For example a show with 1800.3 seconds (30 minutes) will be seeked to 1800 seconds and the end will be reached properly. Or 1799.6 seeked to 1799.

**Comment by drinfernoo** [comment]
This should be fixed in 2.0.17.
